### ISSUE TITLE
Use logger instead of `p` the responses

### ISSUE THREAD

**Issue by bararchy** [raydf/smtp.cr#6]
It seems there are prints to STDOUT hardcoded in the SMTP client class, do you mind allowing the user to pass in a `Logger` class and use it ? You could default to using `Logger.new(STDOUT)` or something

**Comment by ray-delossantos** [comment]
@bararchy, that's correct, right now i don't have the time for making this changes, but PR are welcomed :).
